jose/constants.py

```diff
@@ -31,3 +32,28 @@ class ALGORITHMS(object):
         ES384: hashlib.sha384,
         ES512: hashlib.sha512,
     }
+
+    KEYS = {}
+
+    def get_key(self, algorithm):
+        from jose.jwk import HMACKey, RSAKey, ECKey
```

Comment:
I'm not a fan of this circular dependency.  I would move this key registration/lookup logic to `jwk.py`.